src/ui/features/onboarding/Success/ReferralCodeWidget/ReferralCodeWidget.tsx

```diff
@@ -0,0 +1,188 @@
+            ['--button-background-hover' as string]: '#f0f0f2',
+          }}
+          onClick={onClick}
+        >
+          <HStack gap={8}>
+            <GiftIcon />
+            <GradientText>
+              <UIText kind="body/accent">Enter Referral Code</UIText>
+            </GradientText>
+          </HStack>
+        </Button>
+        <UIText
+          kind="small/regular"
+          color="var(--always-white)"
+          style={{ opacity: 0.5 }}
+        >
+          if you have one
+        </UIText>
+      </VStack>
+    </>
+  );
+}
+
+function ReferralCodeForm({
+  onSuccess,
+}: {
+  onSuccess: (pendingReferrer: ReferrerData) => void;
+}) {
+  const [referralCode, setReferralCode] = useState('');
+
+  const { mutate: applyReferralCode, ...applyReferralCodeMutation } =
+    useMutation({
+      mutationFn: async ({
+        referralCode: rawReferralCode,
+      }: {
+        referralCode: string;
+      }) => {
+        const referralCode = rawReferralCode.trim();
+
+        // TODO: Get my referral code & check if we're referring ourselves
+
+        // if (myReferralCode === referralCode) {
+        //   throw new Error('Can not apply your own referral code');
+        // }
+
+        const referrer = await walletPort.request(
+          'uiApplyReferralCodeToAllWallets',
+          {
+            referralCode,
+          }
+        );
+        onSuccess(referrer);
+      },
+    });
+
+  return (
+    <>
+      <Spacer height={28} />
+      <VStack gap={16} className={styles.referralCodeForm}>
+        <HStack gap={8} alignItems="center">
+          <UIText kind="body/accent" color="var(--always-white)">
+            Referral Code
+          </UIText>
+          <div
+            style={{
+              cursor: 'help',
+              display: 'flex',
+              color: 'var(--always-white)',
+            }}
+            title="Your Referral Code"
+          >
+            <QuestionIcon style={{ width: 24, height: 24 }} />
+          </div>
+        </HStack>
+
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            const form = event.currentTarget;
+            if (!form.checkValidity()) {
+              return;
+            }
+            const formData = collectData(form, {});
+            const referralCode = formData.referralCode as string;
+
+            applyReferralCode({ referralCode });
+          }}
+        >
+          <HStack gap={8} alignItems="center">
+            <ZStack>
+              <Input
+                className={styles.referralCodeInput}
+                name="referralCode"
+                autoFocus={true}
+                boxHeight={40}
+                placeholder="Enter code"
+                required={true}
+                value={referralCode}
+                onChange={(event) => setReferralCode(event.target.value)}
+                disabled={applyReferralCodeMutation.isLoading}
+              />
+              {referralCode.length > 0 ? (
```

Comment:
We can probably use `input:not([value=""])` css selector here and not use useState() at all. As I can see, we don't use it anywhere else

Reply:
However, not critical